dtypes.py

```diff
@@ -153,6 +153,8 @@ def fill_registry_with_c_types(reg, respect_windows, include_bool=True):
 
     reg.get_or_register_dtype("float", np.float32)
     reg.get_or_register_dtype("double", np.float64)
+    reg.get_or_register_dtype("complex", np.complex64)
```

Comment:
I'm not a fan. This will break OpenCL, which reserves but does not define the complex types. At best, I could see adding a flag (like for `bool`, also for OpenCL). But the more sustainable thing is to have each loopy target build up its own type registry, with C-with-complex as a specific subclass.

Reply:
Where do I look to do this properly?

Reply:
Override [this](https://github.com/inducer/loopy/blob/15db4baea1deaebcad99e4a571b8ef6e594961b9/loopy/target/c/__init__.py#L351) in a subclass.

Reply:
Added the complex side of things in #30.